src/crafting_gui.cpp

```diff
@@ -48,6 +51,40 @@ void load_recipe_category( JsonObject &jsobj )
     }
 }
 
+std::string get_cat_name( std::string prefixed_name )
+{
+    if( prefixed_name.find("CC_") == 0 ) {
+        return prefixed_name.substr( 3, prefixed_name.size() - 3 );
+    }
+
+    debugmsg( ( "could not get screen name of: " + prefixed_name ).c_str() );
+    return prefixed_name;
+}
+
+std::string get_subcat_name( const std::string &cat, std::string prefixed_name )
+{
+    std::string prefix = "CSC_" + get_cat_name( cat ) + "_";
+
+    if( prefixed_name.find( prefix ) == 0 ) {
+        return prefixed_name.substr( prefix.size(), prefixed_name.size() - prefix.size() );
+    } else if( prefixed_name.find("CSC_ALL") == 0 ) {
+        return "ALL";
+    }
+
+    debugmsg( ( "could not get screen name of: " + prefixed_name ).c_str() );
```

Comment:
It might be better to check the prefix when the categories/subcategories get loaded from JSON. At that point one can throw an error if the string does have the expected format (starting with "CSC_"). This gives the user a fast feedback if their changes don't match the format, including the location of the problem in the JSON file.

Afterwards (in this function), one can simply assume that the strings have the required prefix and remove it without checking. This removes the need for the debug message.